### PR TITLE
Elaborate Contract Basics based on new developer experience

### PR DESCRIPTION
**Pull request by mudassir-agoric**
This pull request includes changes to the `contract-basics.md` file, which generates the documentation page titled `Smart Contract Basics`. Here's a summary of the modifications:

1. I have revised the language used in the section introducing the greetings smart contract. The wording is now more formal and easier to follow.
2. I've added additional text to the testing section of the contract basics to enhance readability.
3. I've removed extraneous code from the `test-zoe-hello.js` file, specifically code unrelated to the greetings smart contract. To maintain existing links and references, I've preserved the original file without alterations. Additionally, I've created a new file named `test-zoe-01-hello.js`, which will be used in the contract basics documentation.
4. The docs in the first part of `contract-basics` links/makes references to only `test-zoe-01-hello.js` file.

### CODE REVIEW

File: main/guides/zoe/contract-basics.md [hunk 1]
```diff
@@ -1,46 +1,56 @@
...
+basic dapp](../getting-started/) in the previous section, let's cover some basics by writing a simple contract that returns a greetings message. We will simply call it _greetings smart contract_. 
 
-A contract is defined by a JavaScript module that exports a `start` function
-that implements the contract's API.
+A contract is defined by a JavaScript module that exports a `start` function. For our greetings smart contract, the declaration of `start` function looks like this:
 
 <<< @/../snippets/zoe/src/01-hello.js#start
 
-Let's start with a contract with a simple `greet` function:
+Apart from `start`, we can have extra functions in our contract. These functions may be defined outside or within the `start` function depending on whether they need access to the variables defined in the `start` function.  For the greetings smart contract, we will have a simple `greet` function (outside `start`) that takes a string as a parameter (for example, name of the person calling the function) and returns a customized greeting message.
```
Comment (toliaqat) [inline main/guides/zoe/contract-basics.md:10]: > In addition to start, we can include additional functions in our contract. These functions can be defined either outside or within the start function, depending on whether they need access to variables defined in start. 

This can be reduced/concise for users who probably might already be familiar with basic JavaScript concepts.

File: main/guides/zoe/contract-basics.md [hunk 2]
```diff
@@ -1,46 +1,56 @@
...
 
 <<< @/../snippets/zoe/src/01-hello.js#greet
 
-The `start` function can expose the `greet` function
-as part of the contract API by making it
-a method of the contract's `publicFacet`:
+The `greet` function, along with any other function besides `start`, must be exposed through the `start` function before it can be called. This is accomplished using a special object called `publicFacet`, which is returned by the `start` function. In the greeting contract, the `start` function exposes the `greet` function by defining it as a method of the contract's `publicFacet`, as shown below:
```
Comment (toliaqat) [inline main/guides/zoe/contract-basics.md:14]: > must be exposed through the `start` function before it can be called.

The paragraph implies that the start function directly exposes other functions through publicFacet. However, the start function is primarily used to initialize the contract and return facets (such as publicFacet and creatorFacet), which can then expose methods like greet.

File: main/guides/zoe/contract-basics.md [hunk 3]
```diff
@@ -1,46 +1,56 @@
...
 
 <<< @/../snippets/zoe/src/01-hello.js#publicFacet
 
-We mark it `Far(...)` to allow callers to use it from outside the contract
-and give it a suggestive interface name for debugging.
+We wrap the reference to `greet` function in a `Far(...)` function call. This will allow callers to use it from outside the contract.
```
Comment (toliaqat) [inline main/guides/zoe/contract-basics.md:18]: > We wrap the reference to `greet` function in a `Far(...)` function call. This will allow callers to use it from outside the contract.

I think it's a more accurate description of Far.

We wrap the greet function in a Far(...) call to safely expose it as a remote object, accessible from outside the contract.

File: main/guides/zoe/contract-basics.md [hunk 4]
```diff
@@ -1,46 +1,56 @@
...
 
 Putting it all together:
 
 <<< @/../snippets/zoe/src/01-hello.js#contract
 
+Let us save this code to a file named `01-hello.js` inside `src` directory. 
 ## Using, testing a contract
 
-Let's use some tests to explore how a contract is used.
+Before we deploy or start our contract, it is usually a good idea to run a few tests that validate its correctness. Agoric contracts are typically tested using the [ava](https://github.com/avajs/ava) framework. They start with `@endo/init` to establish a [Hardened JavaScript](../js-programming/hardened-js) environment as below:
```
Comment (toliaqat) [inline main/guides/zoe/contract-basics.md:29]: Replace `They start with @endo/init...` with `Tests start with @endo/init...`

Replace `...few tests that validate...` with `...few tests to validate...`

File: main/guides/zoe/contract-basics.md [hunk 5]
```diff
@@ -1,46 +1,56 @@
...
 
-Agoric contracts are typically tested
-using the [ava](https://github.com/avajs/ava) framework.
-They start with `@endo/init` to establish a [Hardened JavaScript](../js-programming/hardened-js) environment:
+<<< @/../snippets/zoe/contracts/test-zoe-01-hello.js#test-imports
 
-<<< @/../snippets/zoe/contracts/test-zoe-hello.js#test-imports
+We also import `E()` in order to make asynchronous method calls and `test` function from `ava`. _We'll talk more about [using `E()` for async method calls](../js-programming/eventual-send) later._ Also note the `eslint-disable-next-line` command in the comment on line 3; this is needed in order to suppress an error that arises due to a known issue in `ava`.
```
Comment (toliaqat) [inline main/guides/zoe/contract-basics.md:33]: command -> comment

File: main/guides/zoe/contract-basics.md [hunk 6]
```diff
@@ -1,46 +1,56 @@
...
+Let's save this code in a file named `test-01-hello.js` in a `test` directory. Both `src` and `test` directories should lie in the same `contract` directory. Let us run the following command to execute the test:
+
+```sh
+npx ava --match="contract greets by name"
+```
+You should see the following line towards the end of the output:
+```
+1 test passed
+```
+Congratulations! You have written and tested your first smart contract. Our next goal is to deploy this contract on the local blockchain and make it more interesting by adding some business logic to it.
```
Comment (toliaqat) [inline main/guides/zoe/contract-basics.md:53]: You can remove "to it" from the end.

File: main/guides/zoe/contract-basics.md (hunk 2 above)
Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md:14]: Would it be fine if we removed "through the `start` function " from the first sentence? Note that the audience probably does not know anything about `facet` at this point.

File: main/guides/zoe/contract-basics.md (hunk 3 above)
Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md:18]: Will fix it.

File: main/guides/zoe/contract-basics.md (hunk 4 above)
Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md:29]: Fixed second one as suggested. Changed first one to `The test file begins with`.

File: main/guides/zoe/contract-basics.md (hunk 6 above)
Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md:53]: Fixed.

File: main/guides/zoe/contract-basics.md (hunk 5 above)
Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md:33]: Are you referring to the `eslint-disable-next-line`?  Let us talk about it. I think it should as is but I will fix it if you think otherwise.

File: main/guides/zoe/contract-basics.md (hunk 1 above)
Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md:10]: Let us talk about it. I am a bit inclined to err towards redundancy in the first tutorial.

File: main/guides/zoe/contract-basics.md (hunk 2 above)
Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md]: After our discussion, I am changing it to the following:

The `greet` function, along with any other public function, must be made accessible through the `publicFacet` of the contract. The `publicFacet` is returned by the `start` function, which initializes the contract. In the greeting contract, the `start` function exposes the `greet` function by defining it as a method of the contract's `publicFacet`, as shown below:

File: main/guides/zoe/contract-basics.md (hunk 1 above)
Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md]: I made it short after our discussion, as below: 

For the greetings smart contract, we will have a simple `greet` function apart from `start` function. The `greet` function takes a string as a parameter (for example, name of the person calling the function) and returns a customized greeting message.

File: main/guides/zoe/contract-basics.md [hunk 7]
```diff
@@ -1,46 +1,56 @@
...
 
 Putting it all together:
 
 <<< @/../snippets/zoe/src/01-hello.js#contract
 
+Let us save this code to a file named `01-hello.js` inside `src` directory. 
 ## Using, testing a contract
 
-Let's use some tests to explore how a contract is used.
+Before we deploy or start our contract, it is usually a good idea to run a few tests to validate its correctness. Agoric contracts are typically tested using the [ava](https://github.com/avajs/ava) framework. The test file begins with an `import @endo/init` to establish a [Hardened JavaScript](../js-programming/hardened-js) environment as below:
```
Comment (dckc) [inline main/guides/zoe/contract-basics.md:29]: TECHNICAL:

Why introduce deployment? Deploying this contract makes little sense. It doesn't have any offer handlers nor publish anything to vstorage, so there's no way to interact with it.

The PR title and description don't mention deployment, so it seems out of scope of this PR.

File: main/guides/zoe/contract-basics.md [hunk 8]
```diff
@@ -1,46 +1,56 @@
...
+Let's save this code in a file named `test-01-hello.js` in a `test` directory. Both `src` and `test` directories should lie in the same `contract` directory. Let us run the following command to execute the test:
+
+```sh
+npx ava --match="contract greets by name"
+```
+You should see the following line towards the end of the output:
+```
+1 test passed
+```
+Congratulations! You have written and tested your first smart contract. Our next goal is to deploy this contract on the local blockchain and make it more interesting by adding some business logic.
```
Comment (dckc) [inline main/guides/zoe/contract-basics.md:53]: TECHNICAL:

Again, any discussion of deployment seems out of scope.

File: main/guides/zoe/contract-basics.md [hunk 9]
```diff
@@ -1,46 +1,56 @@
...
-Let's use some tests to explore how a contract is used.
+Before we deploy or start our contract, it is usually a good idea to run a few tests to validate its correctness. Agoric contracts are typically tested using the [ava](https://github.com/avajs/ava) framework. The test file begins with an `import @endo/init` to establish a [Hardened JavaScript](../js-programming/hardened-js) environment as below:
 
-Agoric contracts are typically tested
-using the [ava](https://github.com/avajs/ava) framework.
-They start with `@endo/init` to establish a [Hardened JavaScript](../js-programming/hardened-js) environment:
+<<< @/../snippets/zoe/contracts/test-zoe-01-hello.js#test-imports
 
-<<< @/../snippets/zoe/contracts/test-zoe-hello.js#test-imports
+We also import `E()` in order to make asynchronous method calls and `test` function from `ava`. _We'll talk more about [using `E()` for async method calls](../js-programming/eventual-send) later._ Also note the `eslint-disable-next-line` comment on line 3; this is needed in order to suppress an error that arises due to a known issue in `ava`.
```
Comment (dckc) [inline main/guides/zoe/contract-basics.md:33]: TECHNICAL:

Why should the reader note the eslint stuff? It seems unrelated to smart contract basics.

If they're curious, they can follow the URL.

I'd rather try to get rid of it than try to explain it here.

File: main/guides/zoe/contract-basics.md [hunk 10]
```diff
@@ -1,46 +1,56 @@
...
-<<< @/../snippets/zoe/contracts/test-zoe-hello.js#test1
+
+Putting it all together:
+
+<<< @/../snippets/zoe/contracts/test-zoe-01-hello.js#test-01-hello
+
+Let's save this code in a file named `test-01-hello.js` in a `test` directory. Both `src` and `test` directories should lie in the same `contract` directory. Let us run the following command to execute the test:
+
+```sh
+npx ava --match="contract greets by name"
```
Comment (dckc) [inline main/guides/zoe/contract-basics.md:47]: TECHNICAL (and **critical** due to security):

Let's not encourage use of `npx`. One typo like `npx aava` and who knows what code is running on their machine.

Comment (dckc) [inline main/guides/zoe/contract-basics.md:47]: The commands to run tests are discussed in the next section
https://docs.agoric.com/guides/zoe/contract-walkthru.html

Does it seem important to repeat them here?

File: main/guides/zoe/contract-basics.md (hunk 8 above)
Comment (dckc) [reply main/guides/zoe/contract-basics.md:53]: EDITORIAL:

Plus, the next goals are to learn about State and Access Control, no?

File: main/guides/zoe/contract-basics.md (hunk 9 above)
Comment (dckc) [reply main/guides/zoe/contract-basics.md:33]: We could skip right to "A test that the `greet` method works..." and leave discussion of test set-up to the next section (complete walkthru).

File: main/guides/zoe/contract-basics.md (hunk 2 above)
Comment (dckc) [reply main/guides/zoe/contract-basics.md]: TECHNICAL:

"The publicFacet is returned by the start function" is incorrect. The start fuction returns an object with a `publicFacet` property.

File: snippets/zoe/contracts/test-zoe-01-hello.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+import test from 'ava';
+// #endregion test-imports
+// #region test1
+import { start } from '../src/01-hello.js';
+
+test('contract greets by name', async t => {
+  const { publicFacet } = start();
+  const actual = await E(publicFacet).greet('Bob');
+  t.is(actual, 'Hello, Bob!');
+});
```
Comment (dckc) [inline snippets/zoe/contracts/test-zoe-01-hello.js:15]: TECHNICAL:
I don't understand why we should maintain all this code in a 2 separate files.

**Review by dckc — CHANGES_REQUESTED** [review]
Please take out the reference to npx.

That's the one critical change request. The others are suggestions. I have tried to separate them into EDITORIAL and TECHNICAL suggestions.

For EDITORIAL, I'd like to hear from others such as @Jovonni and @kbennett2000 .

**Review by kbennett2000 — COMMENTED** [review]
Love the direction and content so far! Is it worth a larger group discussion to decide whether to include deployment at this stage or not?

File: main/guides/zoe/contract-basics.md (hunk 7 above)
Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md:29]: @kbennett2000 the focus here is on explaining the deployment process, using a basic contract primarily as a means to that end, rather than the other way around.

File: main/guides/zoe/contract-basics.md (hunk 2 above)
Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md]: Fixed!

File: main/guides/zoe/contract-basics.md [hunk 12]
```diff
@@ -1,46 +1,56 @@
 # Smart Contract Basics
 
 Before we look at how to make a contract such as the one in [the
-basic dapp](../getting-started/) in the previous section, let's cover some basics.
+basic dapp](../getting-started/) in the previous section, let's cover some basics by writing a simple contract that returns a greetings message. We will simply call it _greetings smart contract_.
```
[the remark this answers is not in the document]
Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md:4]: I would like to, politely, disagree :-) I think the requested change do make life easier for a new dev.

File: main/guides/zoe/contract-basics.md (hunk 9 above)
Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md]: Fixed.

File: main/guides/zoe/contract-basics.md (hunk 10 above)
Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md]: Fixed.

Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md]: I do not see any significant overlap. Although, contract-walkthru needs a lot of fixing IMO but we will get to it once we have sorted out this and #1104.

File: snippets/zoe/contracts/test-zoe-01-hello.js (hunk 11 above)
Comment (mudassir-agoric) [reply snippets/zoe/contracts/test-zoe-01-hello.js:15]: I think it is neat and structured this way and rather easier to maintain.

File: main/guides/zoe/contract-basics.md [hunk 13]
```diff
@@ -1,46 +1,56 @@
...
 
-_We'll talk more about [using `E()` for async method calls](../js-programming/eventual-send) later._
+Following these `import` statements, we write a simple test that validates that the `greet` method works as expected as below:
 
-A test that the `greet` method works as expected looks like:
+<<< @/../snippets/zoe/contracts/test-zoe-01-hello.js#test1
 
-<<< @/../snippets/zoe/contracts/test-zoe-hello.js#test1
+
+Putting it all together:
```
Comment (toliaqat) [inline main/guides/zoe/contract-basics.md:40]: I think "Putting it all together:" can be done in one go instead of showing the test code two times in same screen.

You can remove the test code (without import) above, and just have the following snippet.

Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md:31]: Fixed.

**Comment by mudassir-agoric** [comment]
@dckc I think I have taken care of most of your comments on this PR. Please review when you get time. Thanks.

File: main/guides/zoe/contract-basics.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (dckc) [reply main/guides/zoe/contract-basics.md:4]: What I heard in office hours is that your position is not just based on your opinion, but on several developer's experience with learning the platform. I can go with that.

Ideally, next time, open an issue with some details about that experience; that issue can be prioritized and serve as justification for a PR such as this one.

File: main/guides/zoe/contract-basics.md [hunk 14]
```diff
@@ -1,46 +1,47 @@
 # Smart Contract Basics
 
 Before we look at how to make a contract such as the one in [the
-basic dapp](../getting-started/) in the previous section, let's cover some basics.
+basic dapp](../getting-started/) in the previous section, let's cover some basics by writing a simple contract that returns a greetings message. We will simply call it _greetings smart contract_.
```
Comment (dckc) [inline main/guides/zoe/contract-basics.md:4]: It's supposed to be a hello world contract. Can we call it that, please?

```suggestion
basic dapp](../getting-started/) in the previous section, let's cover some basics by writing a simple hello-world contract. 
```

File: main/guides/zoe/contract-basics.md [hunk 15]
```diff
@@ -1,46 +1,47 @@
 # Smart Contract Basics
 
 Before we look at how to make a contract such as the one in [the
-basic dapp](../getting-started/) in the previous section, let's cover some basics.
+basic dapp](../getting-started/) in the previous section, let's cover some basics by writing a simple contract that returns a greetings message. We will simply call it _greetings smart contract_. 
 
-A contract is defined by a JavaScript module that exports a `start` function
-that implements the contract's API.
+A contract is defined by a JavaScript module that exports a `start` function. For our greetings smart contract, the declaration of `start` function looks like this:
```
Comment (dckc) [inline main/guides/zoe/contract-basics.md:6]:
```suggestion
A contract is defined by a JavaScript module that exports a `start` function. For our hello-world smart contract, the declaration of `start` function looks like this:
```

and likewise throughout

File: main/guides/zoe/contract-basics.md [hunk 16]
```diff
@@ -1,46 +1,47 @@
...
 
 <<< @/../snippets/zoe/src/01-hello.js#start
 
-Let's start with a contract with a simple `greet` function:
+For the greetings smart contract, we will have a simple `greet` function apart from `start` function. The `greet` function takes a string as a parameter (for example, name of the person calling the function) and returns a customized greeting message.
```
Comment (dckc) [inline main/guides/zoe/contract-basics.md:10]: Our audience is expected to be able to read JS. I'm not sure what the "The `greet` function ... returns a ... message" sentence tells them about our smart contract platform.

File: main/guides/zoe/contract-basics.md [hunk 17]
```diff
@@ -1,46 +1,47 @@
...
-The `start` function can expose the `greet` function
-as part of the contract API by making it
-a method of the contract's `publicFacet`:
+The `greet` function, along with any other public function, must be made accessible through the `publicFacet` of the contract. The `start` function returns an object with a `publicFacet` property. In the greeting contract, the `start` function exposes the `greet` function by defining it as a method of the contract's `publicFacet`, as shown below:
 
 <<< @/../snippets/zoe/src/01-hello.js#publicFacet
 
-We mark it `Far(...)` to allow callers to use it from outside the contract
-and give it a suggestive interface name for debugging.
+We wrap the `greet` function in a `Far(...)` call to safely expose it as a remote object, accessible from outside the contract.
```
Comment (dckc) [inline main/guides/zoe/contract-basics.md:18]: TECHNICAL:

```suggestion
We wrap the value of the `publicFacet` property in a `Far(...)` call to safely expose it as a remote object, accessible from outside the contract.
```

In the text before this change, the "it" in "We mark it Far" referred to the public facet object, not the greet function.

File: main/guides/zoe/contract-basics.md [hunk 18]
```diff
@@ -1,46 +1,47 @@
...
+We also give it a suggestive interface name `Hello` for debugging.
```
Comment (dckc) [inline main/guides/zoe/contract-basics.md:19]: TECHNICAL:

This phrasing suggests that giving a suggestive name is separate from wrapping with Far.

```suggestion
This also gives it a suggestive interface name `Hello` for debugging.
```

File: main/guides/zoe/contract-basics.md [hunk 19]
```diff
@@ -1,46 +1,47 @@
...
 
 Putting it all together:
 
 <<< @/../snippets/zoe/src/01-hello.js#contract
 
+Let us save this code to a file named `01-hello.js` inside `src` directory. 
 ## Using, testing a contract
 
-Let's use some tests to explore how a contract is used.
+Before we deploy or start our contract, it is usually a good idea to run a few tests to validate its correctness. Agoric contracts are typically tested using the [ava](https://github.com/avajs/ava) framework. The test file begins with an `import @endo/init` to establish a [Hardened JavaScript](../js-programming/hardened-js) environment. We also import `E()` in order to make asynchronous method calls and `test` function from `ava`. _We'll talk more about [using `E()` for async method calls](../js-programming/eventual-send) later._ Following these `import` statements, we write a simple test that validates that the `greet` method works as expected.
```
Comment (dckc) [inline main/guides/zoe/contract-basics.md:29]: Again, deployment seems out of scope.

File: snippets/zoe/contracts/test-zoe-01-hello.js (hunk 11 above)
Comment (dckc) [reply snippets/zoe/contracts/test-zoe-01-hello.js:15]: I'd like to understand what prompted moving the code in the 1st place.
But I guess I don't feel strongly about it.

File: main/guides/zoe/contract-basics.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md]: Fixed.

File: snippets/zoe/contracts/test-zoe-01-hello.js (hunk 11 above)
Comment (mudassir-agoric) [reply snippets/zoe/contracts/test-zoe-01-hello.js:15]: Well, the file contained tests for unrelated contracts.

File: main/guides/zoe/contract-basics.md (hunk 14 above)
Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md]: Thank you for saying that. Fixed.

File: main/guides/zoe/contract-basics.md (hunk 15 above)
Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md]: Fixed.

File: main/guides/zoe/contract-basics.md (hunk 16 above)
Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md]: It is one line summary of the contract - can we keep it?

File: main/guides/zoe/contract-basics.md (hunk 17 above)
Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md]: Fixed.

File: main/guides/zoe/contract-basics.md (hunk 18 above)
Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md]: Fixed.

File: main/guides/zoe/contract-basics.md (hunk 19 above)
Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md]: Removed. Fixed.

File: main/guides/zoe/contract-basics.md (hunk 16 above)
Comment (dckc) [reply main/guides/zoe/contract-basics.md]: ok

**Review by dckc — APPROVED** [review]
very nice work!

File: main/guides/zoe/contract-basics.md [hunk 20]
```diff
@@ -1,46 +1,46 @@
...
 
-_We'll talk more about [using `E()` for async method calls](../js-programming/eventual-send) later._
+<<< @/../snippets/zoe/contracts/test-zoe-01-hello.js#test-01-hello
 
-A test that the `greet` method works as expected looks like:
+Let's save this code in a file named `test-01-hello.js` in a `test` directory. Both `src` and `test` directories should lie in the same `contract` directory. Let us run the following command to execute the test:
 
-<<< @/../snippets/zoe/contracts/test-zoe-hello.js#test1
+```sh
+ava --match="contract greets by name"
```
Comment (dckc) [inline main/guides/zoe/contract-basics.md:37]: oops! when I requested taking out `npx`, I should have offered a replacement:

```suggestion
yarn ava --match="contract greets by name"
```

I don't expect that invoking `ava` on its own will work unless somebody has done `npm install -g ava`.

not critical, I suppose

Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md:37]: Thanks @dckc. I am adding "follow the instructions there to install ava" right next to where we provide link for "ava". Probably we will fix this with something better soon.

File: main/guides/zoe/contract-basics.md [hunk 21]
```diff
@@ -25,7 +25,7 @@ Putting it all together:
...
 ## Using, testing a contract
 
-Agoric contracts are typically tested using the [ava](https://github.com/avajs/ava) framework. The test file begins with an `import @endo/init` to establish a [Hardened JavaScript](../js-programming/hardened-js) environment. We also import `E()` in order to make asynchronous method calls and `test` function from `ava`. _We'll talk more about [using `E()` for async method calls](../js-programming/eventual-send) later._ Following these `import` statements, we write a simple test that validates that the `greet` method works as expected.
+Agoric contracts are typically tested using the [ava](https://github.com/avajs/ava) framework -  follow the instructions there to install `ava`. The test file begins with an `import @endo/init` to establish a [Hardened JavaScript](../js-programming/hardened-js) environment. We also import `E()` in order to make asynchronous method calls and `test` function from `ava`. _We'll talk more about [using `E()` for async method calls](../js-programming/eventual-send) later._ Following these `import` statements, we write a simple test that validates that the `greet` method works as expected.
```
Comment (dckc) [inline main/guides/zoe/contract-basics.md:28]: Why "follow the instructions there to install `ava`"? In our getting started flow, `ava` is installed with other dev tools in the [yarn install step](https://docs.agoric.com/guides/getting-started/#installing-dapp-dependencies).

Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md:28]: I thought we may not assume that they have a `packages.json` file in their working directory?

Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md:28]: Should I change it to `yarn ava` ?

Comment (dckc) [reply main/guides/zoe/contract-basics.md:28]: No `package.json`? Then how are dependencies such as `import '@endo/init';` satisfied? ES6 module `import` declarations won't even work without `"type": "module"` in `package.json`, right?

This is why I postponed details on how to run a test to the context of dapp-offer-up.

I didn't intend "contract basics" to be a tutorial where you could run the pieces. It was just a bit of background info between the getting-started tutorial and the contract walkthru.

Comment (dckc) [reply main/guides/zoe/contract-basics.md:28]: Yes, if we're going to give a command, `yarn ava` is a good one to give.

Comment (mudassir-agoric) [reply main/guides/zoe/contract-basics.md]: @dckc Pardon my ignorance in the above comment - they will of course have a `package.json`. I have updated the said command to `yarn ava` and removed the sentence: _"follow the instructions there to install `ava`"_. Thanks for your extensive and timely input.
